### PR TITLE
Fix change binding indices

### PR DESCRIPTION
**Pull request by ekcoh**
### Description

Extension of fix submitted by @steinbitglis in https://github.com/Unity-Technologies/InputSystem/pull/1352.

### Changes made

Fixed incorrect indexing when searching for bindings when changing existing bindings and added missing unit tests.

### Checklist

Before review:

- [ ] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

### CODE REVIEW

File: Assets/Tests/InputSystem/CoreTests_Actions_Rebinding.cs [hunk 1]
```diff
@@ -1332,4 +1332,133 @@ public void Actions_CanSaveAndLoadRebinds()
...
+        actionMap.AddBinding("<Keyboard>/c", firstActionInMap);  // first, 1
+        actionMap.AddBinding("<Keyboard>/d", secondActionInMap); // second, 1
+        actionMap.AddBinding("<Keyboard>/e", secondActionInMap); // second, 2
+
+        var accessor = secondActionInMap.ChangeBinding(binding1.binding.name);
+        Assert.IsTrue(accessor.valid);
+        secondActionInMap.ChangeBinding(new InputBinding("<Keyboard>/e")).WithPath("<Keyboard>/f");
+        Assert.AreEqual(2, actionMap.FindBinding(new InputBinding("<Keyboard>/f"), out _)); // exact match
+    }
+}
```
Comment (Rene-Damm) [inline Assets/Tests/InputSystem/CoreTests_Actions_Rebinding.cs]: Needs formatter run.

File: Assets/Tests/InputSystem/CoreTests_Actions_Rebinding.cs [hunk 2]
```diff
@@ -1332,4 +1332,133 @@ public void Actions_CanSaveAndLoadRebinds()
...
+        var binding1 = actionMap.AddBinding("<Keyboard>/a", firstActionInMap);  // first, 0
+        actionMap.AddBinding("<Keyboard>/b", secondActionInMap); // second, 0
+        actionMap.AddBinding("<Keyboard>/c", firstActionInMap);  // first, 1
+        actionMap.AddBinding("<Keyboard>/d", secondActionInMap); // second, 1
+        actionMap.AddBinding("<Keyboard>/e", secondActionInMap); // second, 2
+
+        var accessor = secondActionInMap.ChangeBinding(binding1.binding.name);
+        Assert.IsTrue(accessor.valid);
+        secondActionInMap.ChangeBinding(new InputBinding("<Keyboard>/e")).WithPath("<Keyboard>/f");
+        Assert.AreEqual(2, actionMap.FindBinding(new InputBinding("<Keyboard>/f"), out _)); // exact match
```
Comment (Rene-Damm) [inline Assets/Tests/InputSystem/CoreTests_Actions_Rebinding.cs]: Recommend use of `Assert.That` for consistency. Used for all existing tests.

File: Packages/com.unity.inputsystem/CHANGELOG.md [hunk 3]
```diff
@@ -46,6 +46,7 @@ however, it has to be formatted properly to pass verification tests.
 - `PlayerInput` no longer logs an error message when it is set to `Invoke UnityEvents` and can't find  an action in the given `.inputactions` asset ([case 1259577](https://issuetracker.unity3d.com/issues/an-error-is-thrown-when-deleting-an-input-action-and-entering-play-mode)).
 - Fixed `HoldInteraction` getting stuck when hold and release happens in same event ([case 1346786](https://issuetracker.unity3d.com/issues/input-system-the-canceled-event-is-not-fired-when-clicking-a-button-for-a-precise-amount-of-time)).
 - Fixed adding an action in the `.inputactions` editor automatically duplicating interactions and processors from the first action in the map.
+- Fixed incorrect indexing in `InputActionsSetupExtensions.ChangeBinding`. Contribution by [Fredrik Ludvigsen](https://github.com/steinbitglis) in [#1348](https://github.com/Unity-Technologies/InputSystem/pull/1352).
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/CHANGELOG.md]: `incorrect indexing` is rather vague. Suggest something like
```
* Fixed `InputActionSetupExtensions.ChangeBinding` modifying binding from a different action than specified. ...
```

**Review by Rene-Damm — APPROVED** [review]
Needs formatter run. Otherwise LGTM.

File: Assets/Tests/InputSystem/CoreTests_Actions_Rebinding.cs (hunk 2 above)
Comment (ekcoh) [reply Assets/Tests/InputSystem/CoreTests_Actions_Rebinding.cs]: Good point, forgot that this was the pattern used here. This have now been updated.

File: Packages/com.unity.inputsystem/CHANGELOG.md (hunk 3 above)
Comment (ekcoh) [reply Packages/com.unity.inputsystem/CHANGELOG.md]: This have now been updated similar to suggestion.

File: Assets/Tests/InputSystem/CoreTests_Actions_Rebinding.cs (hunk 1 above)
Comment (ekcoh) [reply Assets/Tests/InputSystem/CoreTests_Actions_Rebinding.cs]: Formatter check now passes.

**Pull request merged**